source/routing/raptor.cpp

```diff
@@ -450,13 +451,15 @@ void RAPTOR::set_valid_jp_and_jpp(
         if (it_commercial_mode != data.pt_data->commercial_modes_map.end()) {
             for (const auto line : it_commercial_mode->second->line_list) {
                 for (auto route : line->route_list) {
-                    for (const auto jp  : route->journey_pattern_list) {
-                        valid_journey_patterns.set(jp->idx, false);
+                    for (const auto& jp_idx: jp_container.get_jps_from_route()[RouteIdx(*route)]) {
+                        valid_journey_patterns.set(jp_idx.val, false);
                     }
                 }
             }
             continue;
         }
+        /*
+        // TODO: physical_mode in jp?
```

Comment:
We have 3 solutions:
* including the physical mode in the JpKey to have one physical mode in a journey pattern
* stop supporting forbidden uri on physical mode
* finding another way to filter on physical mode

What do you think?